conseil-api/src/test/scala/tech/cryptonomic/conseil/api/routes/platform/data/ethereum/EthereumDataOperationsTest.scala

```diff
@@ -94,6 +94,79 @@ class EthereumDataOperationsTest
+        dbHandler.run(Tables.Transactions ++= transactions).isReadyWithin(5.seconds) shouldBe true
+        dbHandler.run(Tables.Receipts ++= receipts).isReadyWithin(5.seconds) shouldBe true
+
+        whenReady(sut.fetchReceipts(Query.empty)) { result =>
+          result.value.size shouldBe 3
+        }
+      }
+
+      "return proper contracts, while fetching all contracts" in {
+        // given
+        dbHandler.run(Tables.Blocks ++= blocks).isReadyWithin(5.seconds) shouldBe true
+        dbHandler.run(Tables.Transactions ++= transactions).isReadyWithin(5.seconds) shouldBe true
+        dbHandler.run(Tables.Contracts ++= contracts).isReadyWithin(5.seconds) shouldBe true
+
+        whenReady(sut.fetchContracts(Query.empty)) { result =>
+          result.value.size shouldBe 3
+        }
+      }
+
+      "return proper tokens, while fetching all tokens" in {
+        // given
+        dbHandler.run(Tables.Blocks ++= blocks).isReadyWithin(5.seconds) shouldBe true
+        dbHandler.run(Tables.Transactions ++= transactions).isReadyWithin(5.seconds) shouldBe true
+        dbHandler.run(Tables.Tokens ++= tokens).isReadyWithin(5.seconds) shouldBe true
+
+        whenReady(sut.fetchTokens(Query.empty)) { result =>
+          result.value.size shouldBe 3
+        }
+      }
+
+      "return proper token transfers, while fetching all token transfers" in {
+        // given
+        dbHandler.run(Tables.Blocks ++= blocks).isReadyWithin(5.seconds) shouldBe true
+        dbHandler.run(Tables.Transactions ++= transactions).isReadyWithin(5.seconds) shouldBe true
+        dbHandler.run(Tables.Tokens ++= tokens).isReadyWithin(5.seconds) shouldBe true
+        dbHandler.run(Tables.TokenTransfers ++= tokenTransfers).isReadyWithin(5.seconds) shouldBe true
+
+        whenReady(sut.fetchTokenTransfers(Query.empty)) { result =>
+          result.value.size shouldBe 3
+        }
+      }
+
+      "return proper number of accounts, while fetching all of accounts" in {
+        // given
+        dbHandler.run(Tables.Blocks ++= blocks).isReadyWithin(5.seconds) shouldBe true
+        dbHandler.run(Tables.Transactions ++= transactions).isReadyWithin(5.seconds) shouldBe true
+
+        whenReady(sut.fetchAccounts(Query.empty)) { result =>
+          result.value.size shouldBe 2
+        }
+      }
+
+      "return proper account by address" in {
+        // given
+        dbHandler.run(Tables.Blocks ++= blocks).isReadyWithin(5.seconds) shouldBe true
+        dbHandler.run(Tables.Transactions ++= transactions).isReadyWithin(5.seconds) shouldBe true
+
+        whenReady(sut.fetchAccountByAddress("to")) { result =>
+          result.value should (contain key "address" and contain value transaction1.to)
+          //TODO Fix test for 'value' field in AccountRow
+          //Issue with the current approach is that type inside case class is from 'scala' while slick is mapping data inside 'QueryResponse' to 'java'.
```

Comment:
Hmmm, what kind of object is returned by the slick response? And exactly why is it a java one?